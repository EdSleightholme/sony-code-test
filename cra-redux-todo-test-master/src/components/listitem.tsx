import React from 'react';

interface IProps {
  title: string;
  done: boolean;
  toDelate: () => void;
}

export const ListItem = ({ title, done, toDelate }: IProps) => {
  return (
    <div>
      {title}
      {done && 'DONE'}
      <div onClick={toDelate}>DELETE TASK </div>
    </div>
  );
};
