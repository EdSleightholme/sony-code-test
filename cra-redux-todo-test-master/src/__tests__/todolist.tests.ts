import { ListItem } from '../components/listitem';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';

describe('ListItem',()=>{

    test('it renders', ()=>{
        const wrapper = shallow(ListItem({title:"title",done:true,toDelate:()=>{}}));
        expect(wrapper).toBeTruthy()
    });

    test('snapshot test',()=>{
        const wrapper = shallow(ListItem({title:"title",done:true,toDelate:()=>{}}));
        expect(wrapper.html()).toMatchSnapshot()
    });
    
    test.todo('displays task name');
    test.todo('take a bool done prop');
    test.todo('function to delete');
})
