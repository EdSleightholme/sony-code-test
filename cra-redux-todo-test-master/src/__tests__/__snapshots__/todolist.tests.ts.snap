// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem snapshot test 1`] = `"<div>titleDONE<div>DELETE TASK </div></div>"`;
